scvid/data/schema.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+from anndata import AnnData
+
+
+class AnnDataSchema:
+    r"""
+    Store reference AnnData attributes for a collection of distributed AnnData objects.
+
+    Validate AnnData objects against reference attributes.
+
+    Example::
+
+        >>> ref_adata = AnnData(X=np.zeros(2, 3))
+        >>> ref_adata.obs["batch"] = [0, 1]
+        >>> ref_adata.var["mu"] = ["a", "b", "c"]
+        >>> adata = AnnData(X=np.zeros(4, 3))
+        >>> adata.obs["batch"] = [2, 3, 4, 5]
+        >>> adata.var["mu"] = ["a", "b", "c"]
+        >>> schema = AnnDataSchema(ref_adata)
+        >>> schema.validate_adata(adata)
+
+    Args:
+        adata (AnnData): Reference AnnData object.
+    """
+
+    attrs = ["obs", "obsm", "var", "varm", "var_names", "layers"]
+
+    def __init__(self, adata: AnnData) -> None:
+        self.attr_values = {}
+        for attr in self.attrs:
+            self.attr_values[attr] = getattr(adata, attr)
```

Comment:
Can you check that `del adata` does not invalidate references to `self.attr_values`?

Reply:
Done.